packages/opencensus-core/src/trace/model/span-base.ts

```diff
@@ -38,7 +38,7 @@ export abstract class SpanBase implements types.Span {
   /** The Span ID of this span */
   readonly id: string;
   /** An object to log information to */
-  logger: Logger;
+  logger!: Logger;
```

Comment:
Similar here on a default logger? (And could giving these defaults allow removing any initialization from the subclasses?)